contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -1454,3 +1467,125 @@ modify_insert_stmt(InsertStmt *stmt, Oid relid)
 	table_close(pg_attribute, AccessShareLock);
 
 }
+
+/*
+ * Stores view object's TSQL definition to bbf_view_def catalog
+ */
+static void
+pltsql_store_view_definition(const char *queryString, ObjectAddress address)
```

Comment:
There can be two corner cases:
1. Creating a view in PG connection with sql_dialect = T-SQL
2. `ALTER VIEW ... ALTER COLUMN` in PG connection

Reply:
1. First case is handled correctly by current implementation.
2. For `ALTER VIEW ... ALTER COLUMN`, We shouldn't allow user to alter TSQL views from pg endpoint. For that I'll add the checks in engine.

Will also add required test cases.

Reply:
If we execute T-SQL in a PG connection, it won't go through ANTLR parser so the result can be different, e.g., `select 'value' = '1';`. Also I just realized that we modify query strings in both ANTLR and parse_analyze stages. Could you check whether we store the original query or modified one? For example,
- `select * from db..object` is changed to `select * from db.dbo.object`
- `select * from dbo.object` is changed to `select * from master_dbo.object`

Reply:
Currently, We've blocked DDL from pg endpoint using GUC.